catalog/core/catalog-core-tagsfilterplugin/src/main/java/org/codice/ddf/catalog/plugin/tagsfilter/TagsFilterQueryPlugin.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.plugin.tagsfilter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.opengis.filter.Filter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.filter.FilterAdapter;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.filter.delegate.TagsFilterDelegate;
+import ddf.catalog.operation.Query;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PreFederatedQueryPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.catalog.source.CatalogProvider;
+import ddf.catalog.source.Source;
+import ddf.catalog.source.UnsupportedQueryException;
+
+/**
+ * When no {@code tags} filter is present on a query, adds a default tag of {@code resource}.
+ * A filter will also be added to include metacards without any tags attribute to support
+ * backwards compatibility.
+ */
+public class TagsFilterQueryPlugin implements PreFederatedQueryPlugin {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TagsFilterQueryPlugin.class);
+
+    private final List<CatalogProvider> catalogProviders;
+
+    private final FilterAdapter filterAdapter;
+
+    private final FilterBuilder filterBuilder;
+
+    public TagsFilterQueryPlugin(List<CatalogProvider> catalogProviders,
+            FilterAdapter filterAdapter, FilterBuilder filterBuilder) {
+        this.catalogProviders = catalogProviders;
+        this.filterAdapter = filterAdapter;
+        this.filterBuilder = filterBuilder;
+    }
+
+    @Override
+    public QueryRequest process(Source source, QueryRequest input)
+            throws PluginExecutionException, StopProcessingException {
+        if (catalogProviders.stream()
```

Comment:
Is this check sufficient? There are places in other repos where sources implement CatalogProvider as well and will not react kindly to having tags added to their queries. If we are going changing those sources in the the other repos then ignore this comment.